### PR TITLE
Reformat runtests.sh

### PR DESCRIPTION
**Pull request by benjamin-gorman**
Co-authored-by: Cameron Trentz <cameron-trentz@uiowa.edu>

### Description
The reformat seeks to improve clarity in what tests have been ran,
are currently being run, and if tests have passed or failed.
The reformat also strives to improve consistency.

- add blue, green, red, bold output
- remove ``MYPYPATH`` from flake8
- fix bug ``"$(which mypy)"`` used for pytype
- fix consistency in segmentation
- fix consistency in separator length
- fix consistency in camelCase usage
- fix consistency in variable naming
- fix consistency in indentation length
- fix consistency in cmdPrefix usage
- fix consistency in equality checks
- fix consistency in whitespace usage
- ~~fix consistency in NUM_PARALLEL with setup.cfg~~

### Status
**Ready**

### Types of changes

- [x] Bug fix

### CODE REVIEW

File: runtests.sh [hunk 1]
```diff
@@ -20,16 +21,13 @@ doUnitTests=true
...
 doPytypeFormat=false
 doMypyFormat=false
-doFlake8Format=false
-clean_cache_files=false
 
-NUM_PARALLEL=1
-
-# testing command to run
-cmd="python3"
+doCleanup=false
```
Comment (hjmjohnson) [inline runtests.sh:28]: This initial value should be at line 27.

File: runtests.sh [hunk 2]
```diff
@@ -20,16 +21,13 @@ doUnitTests=true
...
 
+NUM_PARALLEL=8
```
Comment (hjmjohnson) [inline runtests.sh:30]: Need something a bit more sophisticated here if default is > 1.  Not everyone has 8 processor to use.

NUM_PARALLEL=$(python3 -c "import multiprocessing; print(f'{multiprocessing.cpu_count()}')")

File: runtests.sh (hunk 1 above)
Comment (benjamin-gorman) [reply runtests.sh:28]: Fixed

File: runtests.sh (hunk 2 above)
Comment (benjamin-gorman) [reply runtests.sh:30]: The purpose was to have it be consistent with ``setup.cfg``.

```cfg
[pytype]
jobs = 8
```

However, ``pytype -j ${NUM_PARALLEL}`` takes precedent over ``setup.cfg`` so the value in ``setup.cfg`` doesn't matter.

Changed it back to ``1``.

**Comment by hjmjohnson** [comment]
Failures introduced by these changes on CI machines.

**Comment by benjamin-gorman** [comment]
> Failures introduced by these changes on CI machines.

```bash
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
##[error]Process completed with exit code 2.
```

I expected this to solve the problem

```bash
[[ -z "$TERM" ]] && export TERM=dumb # if there is no value for $TERM use dumb
```

I also tried this way

```bash
blue="$([[ -n "$TERM" ]] && tput bold; [[ -n "$TERM" ]] && tput setaf 4)"
green="$([[ -n "$TERM" ]] && tput bold; [[ -n "$TERM" ]] && tput setaf 2)"
red="$([[ -n "$TERM" ]] && tput bold; [[ -n "$TERM" ]] && tput setaf 1)"
noColor="$([[ -n "$TERM" ]] && tput sgr0)"
```

**Pull request merged**